uk.co.compucorp.civicrm.hrcore/backstop_data/page-objects/ssp-reports.js

```diff
@@ -0,0 +1,24 @@
+var modal = require('./page');
+
+module.exports = modal.extend({
+  /**
+   * The Iframe embedding the SSP reports in the CiviCRM admin adapts its height
+   * dynamically whenever the viewport resizes. This can lead to false positives
+   * due to the resize that headless Chrome performs just before taking the screenshot
+   * (the iframe would not resize consistently every time, leading BackstopJS to
+   * report height differences of a handful of pixels)
+   *
+   * In order to avoid false positives, the height of the iframe is fixed
+   * by applying a generated-on-the-fly style to it, so that the height can't
+   * change when Chrome resizes the viewport.
+   */
+  waitForReady: function () {
+    this.chromy.evaluate(function () {
+      var tempStyle = document.createElement('style');
+      tempStyle.type = 'text/css';
+      tempStyle.innerHTML = '#reportsIframe { height: 1000px !important; }';
```

Comment:
Why specifically 1000px? What would happen if we have something with a greater height inside?

Reply:
> Why specifically 1000px

It's the ~(current height) of the reports iframe currently.

When you start clicking around in it or if you resize the window you will see the iframe changing the height just slightly to adapt (no idea why the click would trigger the resize, but that's what happens), which is what triggers all the false positives. So I had to fix it to the current height to make sure it would not automatically add/remove a bunch of pixels

> What would happen if we have something with a greater height inside?

There isn't going to, based on the current Iframe content. The filters list has a fixed height and it is scrollable and there is nothing else pushing for more vertical room

![reports](https://user-images.githubusercontent.com/6400898/38020646-16eefcf2-327b-11e8-8e38-708099c3b195.png)

In any case this is a trade off: I had to enforce the height (which means we would lose to ability to detect changes in height, in the rare cases they would occur) in order to get rid of the false positives that would make impossible for BackstopJS to return a "success" report.